### ISSUE TITLE
set DirSheet name to the name of the current directory

### ISSUE THREAD

**Issue by anjakefala** [saulpw/visidata#1775]
Stems from #1768

**Comment by anjakefala** [comment]
@geekscrapy I have a change that does this! I just need to figure out an approach for when a subfolder has the same name as its parent folder (e.g. `visidata/visidata`).

Maybe the DirSheet names should nest (sheet 1: `visidata`, press enter on the `visidata` subfolder. sheet 2: `visidata_visidata`.)

**Comment by geekscrapy** [comment]
Yea, I think the name should always be relative to the first directory that was opened

**Comment by anjakefala** [comment]
Okay, I made both changes! Please let me know how things feel/look. =)

**Comment by geekscrapy** [comment]
Ok, the sheet name makes sense, however when saving a sheet the suggested filename is "`.`" (which is restricted on Linux as it is the cwd)

So a suggestion would be to use the same sheet.name as the suggested saveable file name

E.g. the current issue

1. `vd . --config=/dev/null --visidata_dir=/dev/null`
2. `save-sheet`
3. `save to: .` ENTER
4. Dialog: `. already exists. overwrite?` Y
5. Error: `IsADirectoryError: [Errno 21] Is a directory: '.'`

Suggested:
1. `vd . --config=/dev/null --visidata_dir=/dev/null`
2. `save-sheet`
3. `save to: {self.name}` (as with updated sheet name)

**Comment by anjakefala** [comment]
I need a genie-label on github for when it takes me 3 wishes to get it right. =)

**Comment by geekscrapy** [comment]
```
                                                                           
                           `  `  -  .  _                                   
       '  _                  `            .                                
        .     -  .  .  .  . - '             .                              
                                             `.                            
           ` .  .         .  .  .  .           .                           
                   `   `              `  `.      `                         
                 . - .                   '         `.           .nMf       
                .      `               '  iXMMMMMM            .MMMM        
           ..       ..   `     .xxnMM' ,nMMMMMMMMMX  `      :MM XM         
            "MMM> <MMMMMi  !MMMMMMMM',MMMMMMMMMMMMMX  `   .MMf xM'         
             `?MM.`MMMMMMMX.MMMMMMMhHMMMMMMMMMMMMMMM<MM  :MMP .MM          
               `Mh ?MMMMMMMMMMMMMMMMMP""""?MMMMMMMMMMMf 4MMM  MM>          
                 Mh ?MMMP`""MMMMMMMMM'.d$$Nu.`?MMMMMMM  MMMf dMM>          
                  "k MMP e$b.'?MMMMf u$$$$$$$b `MMMMMMMM`MM  `MM>          
                   ?  M d$$$$b.`MMM '$$$$$$$$$b 'MMMMMMM 'M <  ML          
                    \ 4 3$$$$$$b`MX $F""$$$$$$$  MMMMMMMM  \dk 4M          
                   . . L'$$F   $ `X $    $$$$$E 4MMMMMMMMM. "f MMk         
              .x*""` .dM. ?$ouu$ 'M "$ed$$$$$$ '" .:dMMMMMM. ?MMMMr        
             MM>4  xMMMhx. `?$$$ 4MM ?$$$$$P  xMMMMMMMMMMMMM  MMMMP        
             ?MM> MMMMMMMMMMMn  dMMMMx`?$".dMMMMMMMMMMMM???M>              
                  'MP""""""""` HMMMMMMM~.dMMMMMMMMP"" .r xMM               
                   `MMM < 'F :MMMMMMMMMMMMMMMP"`  d$ JMf MMM .             
                 MM 'M :M '  MMMMMMMMMMMMnnndM"  d$F MM  4M  M:            
               .MMMM  .MM  L 4MMMMMMMMP"MMP" .z$ $$k MM> 'M 4MM:           
              .MMMMP .MMM  $r'MMMMMMP  ` .zd$$$$ $$k MMX  ? 4MMM:          
             MMMMMf XMMMf 4$$ 'MMMM  $$$ 3$$$$$$k""  4MM.   'MMMMh         
           .MMMMP  dMMMM   ^" o "" u$$$$ 4??""        MMMr  `MMMMMM:       
          dMMMMf :MMMMM"         `""``           ...; 'MMM:  `MMMMMML      
        xMMMMP" :MMMMMf               ..;;;  .i!!!!!!!  MMMM   "MMMMMM:    
      xMMMMMf  :MMMP""             <!!!!!` !!!!!!!'`  r MMMMh   `MMMMMMh   
    :MMMMMP   xMMM".u.          :!!!!!!!!i!!''` .u  $$F MMMMMh    ?MMMMMMx 
  .dMMMMM"   XMMP  $$$$h .     '''''''```  .ue$$$$R $F.MMMMMMMM    ?MMMMMMM 
 MMMMMMM     MMM. '?$$$'J$$$$$$bbr.d$$$$$$;4$$$$$$$  :MMMMMMMMM     MMMMMMM 
 MMMMMP     MMMMMMM:  " $$$$$$$$$'8$$$$$$$k`$$$P" .nMMMMMMMMMMM     MMMMMMM 
 MMMMM      MMMMMMMMMMn:. `"???$$ $$$$$????'" .xdMMMMMMMMMMMMM"     MMMMMMM 
 MMMM'     `MMMMMMMMMMMMMMMMnn...........ndMMMMMMMMMMMMMMMMMM       MMMMMMM 
 MMMM      `MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM"       XMMMMMMM 
 MMMM       MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM"        nMMMMMMMM 
 MMMMk      `MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMP"          MMMMMMMMMM 
 MMMMMM.      "MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMP"            uMMMMMMMMMMM 
 MMMMMMMM.      `""MMMMMMMMMMMMMMMMMMMMMMMMMP""             .dMMMMMMMMMMMMM 
 MMMMMMMMMh:         `""""???MMMMMMMPP"""                .nMMMMMMMMMMMMMMMM 
 MMMMMMMMMMMMx.                                      .xMMMMMMMMMMMMMMMMMMMM 
 MMMMMMMMMMMMMMMr                                :dMMMMMMMMMMMMMMMMMMMMMMMM 
 MMMMMMMMMMMP"`                         ..:nnMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
 MMMMMMM""                  ...xnnnnHMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM 
 MMMM"         ....ndMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM 
 MMM     xnMMMM""`"""MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM 
 MMM   'MMMMMM     n  'MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM 
 MMM.   "?MMMMMh:.xM>  :MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM 
 MMMMMn    '""???""   :MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM 
 MMMMMMMMhnx.......nHMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
```

**Comment by anjakefala** [comment]
As far as I can tell, this would require setting a custom `getDefaultSaveName` in `DirSheet`:

```
def getDefaultSaveName(sheet):                                                                                                                             
        return sheet.name + '.' + sheet.options.save_filetype
```

Because the standard behaviour in `getDefaultSaveName` is to use `str(source)` if `source` is a `Path`. I created other worse problems when I tried to alter `'.'`  on the `Path` level (e.g. converting it to its `absolute()` version broke other code, and made the directory column feel cluttered). If @saulpw thinks overriding `getDefaultSaveName` on `DirSheet` is a horrible approach, I will try seeing if I can figure something else out.
